satpy/tests/reader_tests/test_atms_sdr_hdf5.py

```diff
@@ -0,0 +1,337 @@
+        self.assertEqual(data_arr.attrs['units'], 'K')
+        self.assertEqual(data_arr.attrs['rows_per_scan'], num_scans)
+        if with_area:
+            self.assertIn('area', data_arr.attrs)
+            self.assertIsNotNone(data_arr.attrs['area'])
+            self.assertEqual(data_arr.attrs['area'].shape, data_arr.shape)
+        else:
+            self.assertNotIn('area', data_arr.attrs)
+
+    def setUp(self):
+        """Wrap HDF5 file handler with our own fake handler."""
+        from satpy.readers.viirs_atms_sdr_utils import JPSS_SDR_FileHandler
+
+        self.reader_configs = config_search_paths(os.path.join('readers', self.yaml_file))
+        # http://stackoverflow.com/questions/12219967/how-to-mock-a-base-class-with-python-mock-library
+        self.p = mock.patch.object(JPSS_SDR_FileHandler, '__bases__', (FakeHDF5_ATMS_SDR_FileHandler,))
+        self.fake_handler = self.p.start()
+        self.p.is_local = True
+
+    def tearDown(self):
+        """Stop wrapping the HDF5 file handler."""
+        self.p.stop()
+
+    def test_init(self):
+        """Test basic init with no extra parameters."""
+        from satpy.readers import load_reader
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([
+            '/path/to/atms/sdr/data/SATMS_j01_d20221220_t0910240_e0921356_b26361_c20221220100456348770_cspp_dev.h5',
+        ])
+        self.assertEqual(len(loadables), 1)
+        r.create_filehandlers(loadables)
+        # make sure we have some files
+        self.assertTrue(r.file_handlers)
+
+    def test_init_start_end_time(self):
+        """Test basic init with start and end times around the start/end times of the provided file."""
+        r = load_reader(self.reader_configs,
+                        filter_parameters={
+                            'start_time': datetime(2022, 12, 19),
+                            'end_time': datetime(2022, 12, 21)
+                        })
+        loadables = r.select_files_from_pathnames([
+            'SATMS_j01_d20221220_t0910240_e0921356_b26361_c20221220100456348770_cspp_dev.h5',
+        ])
+        self.assertEqual(len(loadables), 1)
+        r.create_filehandlers(loadables)
+        # make sure we have some files
+        self.assertTrue(r.file_handlers)
+
+    def test_load_all_bands(self):
+        """Load brightness temperatures for all 22 ATMS channels."""
+        from satpy.readers import load_reader
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([
+            'SATMS_j01_d20221220_t0910240_e0921356_b26361_c20221220100456348770_cspp_dev.h5',
+            'GATMO_j01_d20221220_t0910240_e0921356_b26361_c20221220100456680030_cspp_dev.h5',
+        ])
+        r.create_filehandlers(loadables)
+        ds = r.load(ATMS_CHANNEL_NAMES)
+        self.assertEqual(len(ds), 22)
+        for d in ds.values():
+            self._assert_bt_properties(d, with_area=True)
```

Comment:
As Codescene says, this test and the next could be refactored together through parametrization.

Reply:
Ahh, yes, fixed.